x-pack/plugins/enterprise_search/server/lib/engines/available_indices.ts

```diff
@@ -11,10 +11,24 @@ export const availableIndices = async (
   client: IScopedClusterClient,
   indices: string[]
 ): Promise<string[]> => {
-  if (await client.asCurrentUser.indices.exists({ index: indices })) return indices;
+  const closedIndicesList = await closedIndices(client);
 
-  const indicesAndExists: Array<[string, boolean]> = await Promise.all(
-    indices.map(async (index) => [index, await client.asCurrentUser.indices.exists({ index })])
+  const indicesExistsAndOpen: Array<[string, boolean]> = await Promise.all(
+    indices.map(async (index) => [
+      index,
+      (await client.asCurrentUser.indices.exists({ index })) && !closedIndicesList.includes(index),
+    ])
   );
-  return indicesAndExists.flatMap(([index, exists]) => (exists ? [index] : []));
+
+  return indicesExistsAndOpen.flatMap(([index, exists]) => (exists ? [index] : []));
+};
+
+const closedIndices = async (client: IScopedClusterClient): Promise<string[]> => {
+  const indexDataResult = await client.asCurrentUser.indices.get({
+    expand_wildcards: ['closed'],
+    features: ['aliases', 'settings'],
+    filter_path: ['*.settings.index.verified_before_close'],
```

Comment:
`verified_before_close` is a internal flag which determines if a index is closed

Reply:
question: Could we just add this flag to the availableIndicesList call below, and then use that to filter out closed indices? That way we don't need to make two superfluous calls.

Reply:
Yes. Didn't get a chance modify this PR yesterday. Added now:  5d3d989244c27624f1417e8fbccfb4e473ed5360